### PR TITLE
Introduce Epinio stack

### PR DESCRIPTION
**Pull request by jimmykarily**
Part of: https://github.com/epinio/helm-charts/issues/63

Opening here for a first epinio-internal review

### CODE REVIEW

**Comment by jimmykarily** [comment]
`shellcheck stacks/epinio/*.sh` is happy

File: stacks/epinio/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+set -e
+
+SCRIPT_DIR="$( cd "$( dirname "$0" )" && pwd )"
+./"${SCRIPT_DIR}/common.sh"
```
Comment (juadk) [inline stacks/epinio/deploy.sh]: it doesn't work on my env:
`./deploy.sh: line 6: .//home/juadk/git/marketplace-kubernetes/stacks/epinio/common.sh: No such file or directory`
```suggestion
source "${SCRIPT_DIR}/common.sh"
```

File: stacks/epinio/upgrade.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+
+set -e
+
+SCRIPT_DIR="$( cd "$( dirname "$0" )" && pwd )"
+./"${SCRIPT_DIR}/common.sh"
```
Comment (juadk) [inline stacks/epinio/upgrade.sh]: same comment as above for loading common.sh

**Review by juadk — CHANGES_REQUESTED** [review]
Tested on k3d, I only had to change traefik namespace to `kube-system` but it was expected for k3d.

Command to source common.sh looks wrong, check my suggestion.

deploy ✅ 
upgrade ✅ 
uninstall ✅

File: stacks/epinio/deploy.sh (hunk 1 above)
Comment (jimmykarily) [reply stacks/epinio/deploy.sh]: `source` was giving me a warning from shellcheck:

```
In deploy.sh line 6:
source "${SCRIPT_DIR}/common.sh"
^-- SC3046 (warning): In POSIX sh, 'source' in place of '.' is undefined.
       ^-----------------------^ SC1091 (info): Not following: ./common.sh was not specified as input (see shellcheck -x).
```

Comment (juadk) [reply stacks/epinio/deploy.sh:6]: and what about:
`. /"${SCRIPT_DIR}/common.sh"`

Comment (juadk) [reply stacks/epinio/deploy.sh]: yes shellcheck only accepts `./"${SCRIPT_DIR}/common.sh"`
Don't know why it fails on my env though.

**Review by juadk — APPROVED** [review]

File: stacks/epinio/common.sh [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+TRAEFIK_CHART_VERSION="10.14.2"
+CERT_MANAGER_CHART_VERSION="1.7.1"
+
+EPINIO_NAMESPACE="epinio"
+TRAEFIK_NAMESPACE="traefik"
+CERT_MANAGER_NAMESPACE="cert-manager"
+
+get_values () {
+  values=""
+  if [ -z "${MP_KUBERNETES}" ]; then
```
Comment (manno) [inline stacks/epinio/common.sh]: nit: curly braces without variable expansion

File: stacks/epinio/deploy.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+set -e
+
+SCRIPT_DIR="$( cd "$( dirname "$0" )" && pwd )"
+. "${SCRIPT_DIR}/common.sh"
+
+update_repos
+
+EPINIO_VALUES="$(get_values)"
```
Comment (manno) [inline stacks/epinio/deploy.sh]: nit: useless quotes

File: stacks/epinio/deploy.sh [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+install_traefik
+
+echo "Installing CertManager"
+install_cert_manager
+
+echo "Waiting for ingress controler IP address"
+traefikIP=$(wait_for_traefik_IP)
+
+echo "Installing $STACK"
+install_epinio "$EPINIO_VALUES" "${traefikIP}.omg.howdoi.website"
```
Comment (manno) [inline stacks/epinio/deploy.sh]: nit: curly braces without variable expansion

File: stacks/epinio/upgrade.sh [hunk 6]
```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+
+set -e
+
+SCRIPT_DIR="$( cd "$( dirname "$0" )" && pwd )"
+. "${SCRIPT_DIR}/common.sh"
```
Comment (manno) [inline stacks/epinio/upgrade.sh]: nit: curly braces without variable expansion

File: stacks/epinio/common.sh [hunk 7]
```diff
@@ -0,0 +1,89 @@
+#!/bin/sh
```
Comment (manno) [inline stacks/epinio/common.sh:1]: Using a shared file is complicated, especially if the script should be runnable from any working dir.

How many of these functions and variables are really shared between install/uninstall/upgrade?
This adds a lot of global state to the individual scripts.

File: stacks/epinio/deploy.sh [hunk 8]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
```
Comment (manno) [inline stacks/epinio/deploy.sh:1]: Does this have to be "deploy", install/uninstall is a better pair.

File: stacks/epinio/common.sh (hunk 7 above)
Comment (jimmykarily) [reply stacks/epinio/common.sh:1]: `deploy.sh` and `upgrade.sh` are almost identical with the difference that `upgrade.sh` doesn't use `--set` for the `global.domain`. I tried other solutions but this one felt cleaner. 

Scripts should be runnable from any dir given how this file is sourced:

```
SCRIPT_DIR="$( cd "$( dirname "$0" )" && pwd )"
. "${SCRIPT_DIR}/common.sh"
```

File: stacks/epinio/common.sh (hunk 3 above)
Comment (jimmykarily) [reply stacks/epinio/common.sh]: removing

File: stacks/epinio/deploy.sh (hunk 8 above)
Comment (jimmykarily) [reply stacks/epinio/deploy.sh:1]: https://github.com/digitalocean/marketplace-kubernetes/blob/master/CONTRIBUTING.md#adding-your-application

File: stacks/epinio/deploy.sh (hunk 4 above)
Comment (jimmykarily) [reply stacks/epinio/deploy.sh]: fixed

File: stacks/epinio/deploy.sh (hunk 5 above)
Comment (jimmykarily) [reply stacks/epinio/deploy.sh]: removed

File: stacks/epinio/upgrade.sh (hunk 6 above)
Comment (jimmykarily) [reply stacks/epinio/upgrade.sh]: removed

**Comment by jimmykarily** [comment]
Reviews are done. Closing this and opening an upstream one.

**Pull request closed without merging**